_drafts/wsl2-firmware-development.md

```diff
@@ -0,0 +1,655 @@
+---
+# This is a template for Interrupt posts. Use previous, more recent posts from the _posts/
+# directory for more inspiration and patterns.
+#
+# When submitting a post for publishing, submit a PR with the post in the _drafts/ directory
+# and it will be moved to _posts/ on the date of publish.
+#
+# e.g.
+# $ cp _drafts/_template.md _drafts/my_post.md
+#
+# It will now show up in the front page when running Jekyll locally.
+
+title: WSL2 for Firmware Development
+description:
+  Article and tutorial about using WSL2 for embedded firmware development.
+author: JP Hutchins
+tags: [zephyr, wsl, wsl2, linux, windows, windows subsystem for linux]
+mermaid: true
+---
+
+<!-- excerpt start -->
+
+The Windows Subsystem for Linux 2 (WSL2) provides a convenient, fast, and stable
+development environment for embedded firmware.  This article discusses
+the pros and cons of WSL2 and presents a complete tutorial for setting up your WSL2
+environment for compiling, flashing, and on-target debugging.
+
+<!-- excerpt end -->
+
+About a year and a half ago, in March of 2023, I took the leap: beginning a new project
+for one of Intercreate's clients, I decided to set up a Zephyr environment in
+WSL2 instead of my trusty VMWare Workstation Linux VM.  Not only did everything go
+smoothly, but since then, many colleagues have also moved from
+native Linux or traditional VMs to WSL2 and have seen excellent outcomes.
+
+{% include newsletter.html %}
+
+{% include toc.html %}
+
+## Features and Limitations
+
+This is a minimal summary intended to guide you on the trade-offs that come with
+the use of various development environments.  It is by no means exhaustive and I'd
+welcome further feedback in the comments!
+
+|                          | Ubuntu 24.04 (Kernel 6.8) | Windows 11 | VMWare Workstation Pro | WSL2 2.2.4.0 (Kernel 5.15) |
+| ------------------------ | ------------------------- | ---------- | ---------------------- | -------------------------- |
+| Graphics                 | ✅                         | ✅          | ✅*                     | ✅                          |
+| Network                  | ✅                         | ✅          | ✅                      | ✅                          |
+| USB                      | ✅                         | ✅          | ✅                      | ✅                          |
+| BLE                      | ✅                         | ✅          | ✅                      | ✅**                        |
+| Performance***           | 100%                      | ~50%       | ?                      | ~88%                       |
+| Developer Experience**** | ⭐⭐⭐⭐                      | ⭐⭐         | ⭐⭐⭐                    | ⭐⭐⭐⭐                       |
```

Comment:
These 4 astericks don't render properly in the .html. I think the column width needs to be wider.

Reply:
![image](https://github.com/user-attachments/assets/fbfc0260-5ee3-43f1-aaf6-ae293dd25a68)

Reply:
Wow, good catch!  I might have to switch to real superscripts, not sure what to do about truncated characters